scripts/base/protocols/modbus/consts.zeek

```diff
@@ -44,13 +47,33 @@ export {
 		[0x85] = "WRITE_SINGLE_COIL_EXCEPTION",
 		[0x86] = "WRITE_SINGLE_REGISTER_EXCEPTION",
 		[0x87] = "READ_EXCEPTION_STATUS_EXCEPTION",
+		[0x88] = "DIAGNOSTICS_EXCEPTION",
```

Comment:
I find the exception codes in function_codes a bit redundant, duplicated.

This does the same locally and allows to remove all the `_EXCEPTION` entries. It's possibly a bit slower, but I'm not worried here. What do you think?

```
--- a/scripts/base/protocols/modbus/main.zeek
+++ b/scripts/base/protocols/modbus/main.zeek
@@ -46,6 +46,25 @@ event zeek_init() &priority=5
        Analyzer::register_for_ports(Analyzer::ANALYZER_MODBUS, ports);
        }
 
+function build_func(func: count): string
+       {
+       local masked = func & ~0x80;
+
+       # If the function code is in function_codes, use it. Also,
+       # if the masked value isn't in function_codes, use function_codes
+       # &default functionality.
+       if ( func in function_codes || masked !in function_codes )
+               return function_codes[func];
+
+       local s = function_codes[masked];
+
+       # Suffix exceptions with _EXCEPTION.
+       if ( func & 0x80 == 0x80 )
+               s += "_EXCEPTION";
+
+       return s;
+       }
+
 event modbus_message(c: connection, headers: ModbusHeaders, is_orig: bool) &priority=5
        {
        if ( ! c?$modbus )
@@ -56,7 +75,7 @@ event modbus_message(c: connection, headers: ModbusHeaders, is_orig: bool) &prio
        c$modbus$ts   = network_time();
        c$modbus$tid = headers$tid;
        c$modbus$unit = headers$uid;
-       c$modbus$func = function_codes[headers$function_code];
+       c$modbus$func = build_func(headers$function_code);
        ## If this message is from the TCP originator, it is a request. Otherwise,
        ## it is a response.
        c$modbus$pdu_type = is_orig ? "REQ" : "RESP";
@@ -65,7 +84,7 @@ event modbus_message(c: connection, headers: ModbusHeaders, is_orig: bool) &prio

```